server.go

```diff
@@ -323,9 +323,21 @@ func (a *App) BatchHandler(w http.ResponseWriter, r *http.Request) {
 		}
 
 		// Object is not found
-		meta, err = a.metaStore.Put(object)
-		if err == nil {
-			responseObjects = append(responseObjects, a.Represent(object, meta, meta.Existing, true, useTus))
```

Comment:
To annotate this for future readers, what's going on here is that if the object is not found, we currently store a stub entry and return an upload response.  However, if we're attempting to download, there's no reason to create a stub, since the user didn't request that, and so we should produce a 404 response for this object instead.

Because the `Put` method returns `meta.Existing` set to true here, we produce both a download and an upload.